packages/tempus/src/Tempus_Stepper_decl.hpp

```diff
@@ -79,6 +80,12 @@ class Stepper
       const Teuchos::RCP<Thyra::ModelEvaluator<Scalar> >& appModel) = 0;
     virtual Teuchos::RCP<const Thyra::ModelEvaluator<Scalar> > getModel() = 0;
 
+    virtual void setAuxModel(
+      const Teuchos::RCP<const AuxModel<Scalar> >& auxModel);
+    virtual void setNonConstAuxModel(
```

Comment:
I can see why you have a separate `setNonConstAuxModel`; it's because the compiler isn't smart enough both to cast to the base class and to cast from nonconst to const at the same time.  I'm not sure if this would increase usability much over just using the existing `rcp_const_cast` function, though.

Reply:
Not sure where I picked up this piece of code, but wouldn't we need 

`RCP<const A> ca_ptr = rcp_implicit_cast<const A>(a_ptr);// To const`

to insert the const, and get rid of `setNonConstAuxModel()`?

Reply:
The following should work:
```c++
RCP<AuxModelSubclass<Scalar>> a_ptr = whatever ();
// implicitly casts the result of rcp_implicit_cast to const AuxModel<Scalar>
thing.setAuxModel (rcp_implicit_cast<AuxModel<Scalar>> (a_ptr));
```